### ISSUE TITLE
Sidecar files only update on the first modification

### ISSUE THREAD

**Issue by clanmills** [Exiv2/exiv2#589]
http://dev.exiv2.org/boards/3/topics/3249?r=3250

**Comment by clanmills** [comment]
As I speculated on Redmine, this is caused by the convertors.  Commenting off the convertors in xmpsidecar.cpp gives rise to loads of issues in the test harness.  So, I've come up with a less aggressive fix in *`XmpSidecar::writeMetadata()`*
```
            // #589 copy XMP tags
            Exiv2::XmpData  copy   ;
            for (Exiv2::XmpData::const_iterator it = xmpData_.begin(); it != xmpData_.end(); ++it) {
                copy[it->key()] = it->value() ;
            }

            // run the convertors
            copyExifToXmp(exifData_, xmpData_);
            copyIptcToXmp(iptcData_, xmpData_);

            // #589 - restore tags which were modified by the convertors
            for (Exiv2::XmpData::const_iterator it = copy.begin(); it != copy.end(); ++it) {
                xmpData_[it->key()] = it->value() ;
            }
``` 

This "calms" the test harness to only report one matter in Testcase 15
```
156a157,161
> Xmp.exif.Flash/exif:Fired                    XmpText     4  True
> Xmp.exif.Flash/exif:Return                   XmpText     1  0
> Xmp.exif.Flash/exif:Mode                     XmpText     1  1
> Xmp.exif.Flash/exif:Function                 XmpText     5  False
> Xmp.exif.Flash/exif:RedEyeMode               XmpText     4  True
164a170
> Exif.Photo.Flash                             Short       1  73
166c172
< Exif.Image.GPSTag                            Long        1  154
---
> Exif.Image.GPSTag                            Long        1  166
```

I think we can fix that.  We should not restore Xmp.exif nor Exif. tags.  We can filter them out when we populate **`copy`** with something such as **`if ( it->key().toString !~ "exif"/i ) copy...`**

I'm unhappy about adding this change at this time in the project.  There is no product stability risk, however it's a semantic change to how we manage sidecars.  This is something to put into Exiv2 v0.28

**Comment by clanmills** [comment]
This is deferred for v0.27.  It's likely to be fixed in v0.27.1 about March 2019.
